lib/expression/eval.ex

```diff
@@ -198,6 +198,33 @@ defmodule Expression.Eval do
   def op(:=, a, b) when is_struct(a, Date) and is_struct(b, Date),
     do: Date.compare(a, b) == :eq
 
+  # credo:disable-for-next-line Credo.Check.Refactor.CyclomaticComplexity
+  def op(operator, a, b)
+      when operator in [:=, :==, :!=, :<, :<=, :>, :>=] and
+             (is_struct(a, Date) or is_struct(a, DateTime)) and
+             is_binary(b) do
```

Comment:
@fedme are we handling a scenario where `a` is a binary and `b` is a struct? Do we need to?

Reply:
In this PR I am only handling the case where the 1st argument is a `Date`/`DateTime` and the 2nd argument is a String. That's all the Canvas needs at the moment but I am happy to expand it to support all combinations